docs/tasks/writingdocs/pushingapatch.rst

```diff
@@ -1,13 +1,185 @@
+Pushing your changes to the VPP Docs Repository
+===============================================
+
+This section will cover how to fork your own branch of the `fdioDocs/vpp-docs <https://github.com/fdioDocs/vpp-docs>`_ repository, clone that repo locally to your computer, make changes to it, see those changes on `Read the Docs <https://readthedocs.org/>`_, and how to issue a pull request when you want your changes to be reflected on the main repo.
+ 
 
 .. toctree::
 
+Forking your own branch 
+------------------------
+In your browser, navigate to the repo you want to branch off of. In this case, the `fdioDocs/vpp-docs <https://github.com/fdioDocs/vpp-docs>`_ repo. At the top right of the page you should see this:
+
+.. figure:: /_images/forkoptions.png
+   :alt: Figure: Repository options on Github 
+   :scale: 50%
+   :align: right
+
+|
+|
+|
+
+Click on "Fork", and then a pop-up should appear where you should then click your Github username. Once this is done, it should automatically take you to the Github page where your new branch is located, just like in the image below.
+
+.. figure:: /_images/usernameforked.png
+   :alt: Figure: Your own branch of the main repo on Github
+   :scale: 35%
+   :align: center
+
+
+Now your **own branch** can be **cloned** to your computer using the URL (https://github.com/YOURUSERNAME/vpp-docs) of the Github page where your branch is located.
+
+
+Creating a local repository
+---------------------------
+
+Now that you have your own branch of the main repository on Github, you can store it locally on your computer. In your shell, navigate to the directory where you want to store your branch/repo. Then execute:
+
+.. code-block:: shell
+
+   $ git clone https://github.com/YOURUSERNAME/vpp-docs
+
+This will create a directory on your computer named **vpp-docs**, the name of the repo. If you want the directory to be named something other than vpp-docs, you can add your own directory name at the end of the command, shown below:
+
+.. code-block:: shell
+
+   $ git clone https://github.com/YOURUSERNAME/vpp-docs OTHERNAME
+
+
+Now that your branch is on your computer, you can modify and build files however you wish.
+
+
+Building required files
+-----------------------
+
+Since we use **.rst** files, **Sphinx**, and **Read the docs** for documenting, you should install `Sphinx <http://www.sphinx-doc.org/en/master/usage/installation.html>`_, and follow their `getting started guide <http://www.sphinx-doc.org/en/master/usage/quickstart.html>`_.
+
+Building these files will generate an **index.html** file, which you can then view in your browser to verify and see your file changes.
+
+Make sure your Sphinx-build directory is **not** the directory of your local repo, or else Git will think you want to stage and commit your build files (which will show up as "Untracked files" when you try to commit). To ensure this, create a new empty directory which will store your build files.
```

Comment:
ok you don't solve this problem with documentation, you solve this problem by creating a gitignore file in the repo (https://git-scm.com/docs/gitignore), see https://git.fd.io/vpp/tree/.gitignore for an example.